pkg/runtime/runtime.go

```diff
@@ -1615,27 +1626,33 @@ func (a *DaprRuntime) getTopicRoutes() (map[string]TopicRoute, error) {
 	}
 
 	for _, s := range subscriptions {
-		if _, ok := topicRoutes[s.PubsubName]; !ok {
-			topicRoutes[s.PubsubName] = TopicRoute{routes: make(map[string]Route)}
+		if topicRoutes[s.PubsubName] == nil {
+			topicRoutes[s.PubsubName] = TopicRoutes{}
 		}
 
-		topicRoutes[s.PubsubName].routes[s.Topic] = Route{metadata: s.Metadata, rules: s.Rules}
-		if len(s.DeadLetterTopic) > 0 {
-			deadLetterTopics[pubsubTopicKey(s.PubsubName, s.Topic)] = s.DeadLetterTopic
+		r := TopicRoute{
+			routes: Route{metadata: s.Metadata, rules: s.Rules},
+		}
+		if s.DeadLetterTopic != "" {
+			r.deadLetterTopic = s.DeadLetterTopic
 		}
+		topicRoutes[s.PubsubName][s.Topic] = r
 	}
 
 	if len(topicRoutes) > 0 {
 		for pubsubName, v := range topicRoutes {
-			topics := []string{}
-			for topic := range v.routes {
-				topics = append(topics, topic)
+			var topics string
+			for topic := range v {
+				if topics == "" {
+					topics += topic
+				} else {
+					topics += " " + topic
+				}
 			}
```

Comment:
[strings.Join](https://pkg.go.dev/strings#Join) might help here

Reply:
Note that I've refactored this again in a new PR that's almost ready